packages/nimble-components/src/table/components/row/tests/table-row.spec.ts

```diff
@@ -240,42 +254,89 @@ describe('TableRow', () => {
             expect(firstCellState).toEqual(firstCell.cellState);
             const firstCellRecord = firstCellState!
                 .cellRecord as TableColumnTextCellRecord;
             expect(firstCellRecord.value).toBe('string 1');
 
             const secondCell = pageObject.getRenderedCell(1)!;
-            const secondCellState = columnStates[1]?.cellState;
+            const secondCellState = cellStates[1];
             expect(secondCellState).toEqual(secondCell.cellState);
             const secondCellRecord = secondCellState!
-                .cellRecord as TableColumnTextCellRecord;
-            expect(secondCellRecord.value).toBe('string 2');
+                .cellRecord as TableColumnDateTextCellRecord;
+            expect(secondCellRecord.value).toBe(0);
         });
 
         it('updates cell.columnId when column changes', async () => {
-            const cell = pageObject.getRenderedCell(0)!;
-            expect(cell.columnId).toEqual('foo');
-            element.removeChild(column1);
+            expect(pageObject.getRenderedCell(0)!.columnId).toEqual('foo');
+            element.removeChild(columnReferences.firstColumn);
             await waitForUpdatesAsync();
-            expect(cell.columnId).toBe('bar');
+            expect(pageObject.getRenderedCell(0)!.columnId).toBe('bar');
         });
 
         it('updates cell.columnId when id of column changes', async () => {
-            const cell = pageObject.getRenderedCell(0)!;
-            expect(cell.columnId).toEqual('foo');
-            column1.columnId = 'baz';
+            expect(pageObject.getRenderedCell(0)!.columnId).toEqual('foo');
+            columnReferences.firstColumn.columnId = 'baz';
             await waitForUpdatesAsync();
-            expect(cell.columnId).toBe('baz');
+            expect(pageObject.getRenderedCell(0)!.columnId).toBe('baz');
+        });
+
+        it('reordering columns reorders cells', async () => {
+            // Swap the two columns
+            element.insertBefore(
+                columnReferences.secondColumn,
+                columnReferences.firstColumn
+            );
+            await waitForUpdatesAsync();
+
+            const cell0 = pageObject.getRenderedCell(0)!;
+            expect(cell0.cellViewTemplate).toEqual(
+                columnReferences.secondColumn.columnInternals.cellViewTemplate
+            );
+            expect(cell0.cellState?.columnConfig).toEqual(
+                columnReferences.secondColumn.columnInternals.columnConfig
+            );
+
+            const cell1 = pageObject.getRenderedCell(1)!;
+            expect(cell1.cellViewTemplate).toEqual(
+                columnReferences.firstColumn.columnInternals.cellViewTemplate
+            );
+            expect(cell1.cellState?.columnConfig).toEqual(
+                columnReferences.firstColumn.columnInternals.columnConfig
+            );
+        });
+
+        it('updating column reuses cell', async () => {
```

Comment:
I'm not a huge fan of these two tests that verify that we recycle cells in some cases vs creating cells in others. That behavior seems like more of an implementation detail, but I struggled to find a way to write a unit test that would catch the issue being fixed by this PR.

Reply:
I'm willing to defer to @rajsite or @jattasNI here. While I agree these tests seem a little too implementation-centric, I can see the potential usefulness of them.

Reply:
No better ideas from me. Perhaps the real assert we want to make for the 'updating column reuses cell' test is actually a performance test that ensures the update rate is still fast. But we're not going to prioritize writing that now so I agree it's worth including these tests.

Reply:
Yea I get the same kind of feeling from some of the virtualizer tests. But agree that it seems worth it to prevent accidental change in behavior.